### PR TITLE
remove extraneous summed potential now that the L-E ixn groups are merged

### PR DESCRIPTION
**Pull request by jkausrelay**
- Remove SummedPotential for SingleTopology class
- Look up NonbondedInteractionGroup directly
- This was missed in https://github.com/proteneer/timemachine/pull/1372

### CODE REVIEW

File: timemachine/fe/free_energy.py [hunk 1]
```diff
@@ -496,16 +496,9 @@ def _combine(self, ligand_values, host_values=None):
 def get_water_sampler_params(initial_state: InitialState) -> NDArray:
     """Given an initial state, return the parameters that define the nonbonded parameters of water with respect to the
     entire system.
-
-    Since we split the different components of the NB into ligand-water, ligand-protein, we want to use the ligand-water parameter
-    to ensure that the ligand component is correctly configured for the specific lambda window.
     """
-    summed_pot = get_bound_potential_by_type(initial_state.potentials, SummedPotential).potential
-    # TBD Figure out a better way of handling the jankyness of having to select the IxnGroup that defines the Ligand-Water
```
Comment (jkausrelay) [inline timemachine/fe/free_energy.py:504]: @badisa  Can you confirm passing in the full NB ixn params (L-W and L-P) to the water sampler is the correct thing to do here? Since we don't split the ixns by type anymore.

Comment (badisa) [reply timemachine/fe/free_energy.py:504]: Hmmmm, would probably be wrong actually. Since the goal of the L-W/P potential is to adjust the water and protein parameters with respect to the Ligand. And currently where we adjust L-W, but not L-P, we are fine since we are moving waters with respect to the ligand. Once we mix them together, moving the waters will be influenced by the changes to L-P since the water will see the protein parameters.

But that is true from https://github.com/proteneer/timemachine/pull/1372 and this PR seems like strictly an improvement. We probably need to move away from this janky behavior where we assume that there is a single potential that contains the right parameters and instead construct a set of water and protein params that are the global state and adjust the ligand parameters independently.

File: tests/test_cuda_targeted_insertion_mover.py [hunk 2]
```diff
@@ -501,8 +501,8 @@ def test_targeted_insertion_brd4_rbfe_with_context(
     box = initial_state.box0
 
     bps = initial_state.potentials
-    summed_pot = get_bound_potential_by_type(initial_state.potentials, SummedPotential)
```
Comment (badisa) [inline tests/test_cuda_targeted_insertion_mover.py:504]: Not for this line: Having tried this change on my own, you will need to unpack the potentials stored in the pickle object used in `test_targeted_insertion_invalid_sample_bug` in this file.

**Comment by proteneer** [comment]
(this PR likely will block #1380 as well)

File: timemachine/fe/free_energy.py (hunk 1 above)
Comment (jkausrelay) [reply timemachine/fe/free_energy.py:504]: Updated the code to handle this and added a test in test_free_energy.py::test_get_water_sampler_params_complex

File: tests/test_cuda_targeted_insertion_mover.py (hunk 2 above)
Comment (jkausrelay) [reply tests/test_cuda_targeted_insertion_mover.py:504]: Updated the pkl using the code shown in https://github.com/proteneer/timemachine/pull/1384/files#diff-1fe4d63075a510de47b049019ef7176bce26001f8bc9b25e1afaaa5c99b9ba49R1080 .

File: tests/test_cuda_bd_exchange_mover.py [hunk 3]
```diff
@@ -932,15 +932,11 @@ def test_bd_moves_with_complex_and_ligand(
     box = initial_state.box0
 
     bps = initial_state.potentials
-    ligand_env_pot = get_bound_potential_by_type(bps, SummedPotential).potential
+    ligand_env_pot = get_bound_potential_by_type(bps, NonbondedInteractionGroup)
```
[the remark this answers is not in the document]
Comment (jkausrelay) [reply tests/test_cuda_bd_exchange_mover.py]: updated

File: tests/test_cuda_targeted_insertion_mover.py [hunk 4]
```diff
@@ -1076,9 +1077,35 @@ def test_targeted_moves_with_complex_and_ligand_in_brd4(
     assert False, f"No moves were made after {iterations}"
 
 
+def update_initial_state(initial_state):
```
Comment (badisa) [inline tests/test_cuda_targeted_insertion_mover.py:1080]: Looks good!

File: tests/test_single_topology.py [hunk 5]
```diff
@@ -791,7 +791,7 @@ def test_combine_with_host():
         potentials.PeriodicTorsion,
         potentials.NonbondedPairListPrecomputed,
         potentials.Nonbonded,
-        potentials.SummedPotential,  # P-L + L-W interactions
+        potentials.NonbondedInteractionGroup,  # P-L + L-W interactions
```
Comment (badisa) [inline tests/test_single_topology.py:794]: nit: to be consistent that it is computing the ligand energies with respect to protein and water (IE which are the row atoms)
```suggestion
        potentials.NonbondedInteractionGroup,  # L-P + L-W interactions
```

File: timemachine/fe/free_energy.py [hunk 6]
```diff
@@ -496,16 +496,20 @@ def _combine(self, ligand_values, host_values=None):
...
-    water_params = summed_pot.params_init[ixn_group_idx]
+    nb_ixn_pot = get_bound_potential_by_type(initial_state.potentials, NonbondedInteractionGroup)
+    water_params = nb_ixn_pot.params
+
+    # If the protein is present, use the original protein parameters for the water sampler
+    if len(initial_state.protein_idxs):
+        prot_params = get_bound_potential_by_type(initial_state.potentials, Nonbonded).params[
+            initial_state.protein_idxs
+        ]
+        if isinstance(water_params, jax.Array):
```
Comment (badisa) [inline timemachine/fe/free_energy.py:508]: Do we need the if/else for jax gradients? If not could we just do `water_params = np.asarray(water_params)[protein_idxs] = ...`?

**Review by badisa — APPROVED** [review]
Could you add benchmarks for this? Don't think this should make much of a difference, but may as we are no longer doing an additional event sync for the summed potential.

**Comment by jkausrelay** [comment]
> Could you add benchmarks for this? Don't think this should make much of a difference, but may as we are no longer doing an additional event sync for the summed potential.

master:
```
dhfr-apo: N=23558 speed: 394.51ns/day dt: 2.5fs (ran 100000 steps in 54.94s)
dhfr-apo-barostat-interval-25: N=23558 speed: 317.31ns/day dt: 2.5fs (ran 100000 steps in 68.26s)
building a protein system with 1758 protein atoms and 7047 water atoms
Optimizing initial state at λ=0.1
----------------------------------------------------------------------
performing BFGS minimization on 265 atoms
(holding the other 8575 atoms frozen)
U(x_0) = -102393.346
Warning: Desired error not necessarily achieved due to precision loss.
         Current function value: -103237.889515
         Iterations: 361
         Function evaluations: 511
         Gradient evaluations: 495
U(x_final) = -103237.890
atom with highest force norm after minimization: 1117
force(x_final)[1117] = [-261.1905371    99.43120306  171.57122914]
----------------------------------------------------------------------
hif2a-apo: N=8805 speed: 701.40ns/day dt: 2.5fs (ran 100000 steps in 30.98s)
hif2a-apo-barostat-interval-25: N=8805 speed: 549.89ns/day dt: 2.5fs (ran 100000 steps in 39.47s)
hif2a-rbfe-barostat-interval-25: N=8840 speed: 505.52ns/day dt: 2.5fs (ran 100000 steps in 42.92s)
hif2a-rbfe-local: N=8840 speed: 1394.09ns/day dt: 2.5fs (ran 100000 steps in 15.68s)
hif2a-rbfe-barostat-interval-25-water-sampling-interval-400: N=8840 speed: 475.14ns/day dt: 2.5fs (ran 100000 steps in 45.66s)
Optimizing initial state at λ=0.1
----------------------------------------------------------------------
performing BFGS minimization on 199 atoms
(holding the other 6118 atoms frozen)
U(x_0) = -83324.835
Warning: Desired error not necessarily achieved due to precision loss.
         Current function value: -84115.276026
         Iterations: 435
         Function evaluations: 552
         Gradient evaluations: 541
U(x_final) = -84115.276
atom with highest force norm after minimization: 645
force(x_final)[645] = [ 232.67846235  -71.46603238 -425.10828159]
----------------------------------------------------------------------
solvent-apo: N=6282 speed: 1113.60ns/day dt: 2.5fs (ran 100000 steps in 19.58s)
solvent-apo-barostat-interval-25: N=6282 speed: 774.76ns/day dt: 2.5fs (ran 100000 steps in 28.07s)
solvent-rbfe-barostat-interval-25: N=6317 speed: 696.87ns/day dt: 2.5fs (ran 100000 steps in 31.19s)
solvent-rbfe-local: N=6317 speed: 1509.83ns/day dt: 2.5fs (ran 100000 steps in 14.49s)
ahfe-barostat-interval-15: N=6316 speed: 705.09ns/day dt: 2.5fs (ran 100000 steps in 30.83s)
ahfe-local: N=6316 speed: 1483.53ns/day dt: 2.5fs (ran 100000 steps in 14.82s)
```

PR 1384:
```
dhfr-apo: N=23558 speed: 400.13ns/day dt: 2.5fs (ran 100000 steps in 54.17s)
dhfr-apo-barostat-interval-25: N=23558 speed: 319.14ns/day dt: 2.5fs (ran 100000 steps in 67.87s)
building a protein system with 1758 protein atoms and 7047 water atoms
Optimizing initial state at λ=0.1
----------------------------------------------------------------------
performing BFGS minimization on 265 atoms
(holding the other 8575 atoms frozen)
U(x_0) = -102393.346
Warning: Desired error not necessarily achieved due to precision loss.
         Current function value: -103237.889515
         Iterations: 361
         Function evaluations: 511
         Gradient evaluations: 495
U(x_final) = -103237.890
atom with highest force norm after minimization: 1117
force(x_final)[1117] = [-261.1905371    99.43120306  171.57122914]
----------------------------------------------------------------------
hif2a-apo: N=8805 speed: 703.66ns/day dt: 2.5fs (ran 100000 steps in 30.88s)
^[[A
hif2a-apo-barostat-interval-25: N=8805 speed: 550.65ns/day dt: 2.5fs (ran 100000 steps in 39.42s)
hif2a-rbfe-barostat-interval-25: N=8840 speed: 505.45ns/day dt: 2.5fs (ran 100000 steps in 42.93s)
hif2a-rbfe-local: N=8840 speed: 1402.57ns/day dt: 2.5fs (ran 100000 steps in 15.59s)
hif2a-rbfe-barostat-interval-25-water-sampling-interval-400: N=8840 speed: 475.38ns/day dt: 2.5fs (ran 100000 steps in 45.65s)
Optimizing initial state at λ=0.1
----------------------------------------------------------------------
performing BFGS minimization on 199 atoms
(holding the other 6118 atoms frozen)
U(x_0) = -83324.835
Warning: Desired error not necessarily achieved due to precision loss.
         Current function value: -84115.276026
         Iterations: 435
         Function evaluations: 552
         Gradient evaluations: 541
U(x_final) = -84115.276
atom with highest force norm after minimization: 645
force(x_final)[645] = [ 232.67846235  -71.46603238 -425.10828159]
----------------------------------------------------------------------
solvent-apo: N=6282 speed: 1116.81ns/day dt: 2.5fs (ran 100000 steps in 19.52s)
solvent-apo-barostat-interval-25: N=6282 speed: 772.58ns/day dt: 2.5fs (ran 100000 steps in 28.15s)
solvent-rbfe-barostat-interval-25: N=6317 speed: 697.83ns/day dt: 2.5fs (ran 100000 steps in 31.15s)
solvent-rbfe-local: N=6317 speed: 1518.08ns/day dt: 2.5fs (ran 100000 steps in 14.41s)
ahfe-barostat-interval-15: N=6316 speed: 704.06ns/day dt: 2.5fs (ran 100000 steps in 30.87s)
ahfe-local: N=6316 speed: 1478.57ns/day dt: 2.5fs (ran 100000 steps in 14.79s)
```
